src/broadcast/scrobblingmanager.cpp

```diff
@@ -0,0 +1,165 @@
+#include <QObject>
+
+#include "broadcast/scrobblingmanager.h"
+#include "control/controlproxy.h"
+#include "engine/enginexfader.h"
+#include "mixer/deck.h"
+
+ScrobblingManager::ScrobblingManager() :
+m_CPGuiTick("[Master]", "guiTick50ms",this),
+m_CPCrossfader("[Master]","crossfader", this),
+m_CPXFaderCurve(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                "xFaderCurve"),this),
+
+m_CPXFaderCalibration(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                      "xFaderCalibration"),this),
+
+m_CPXFaderMode(ConfigKey(EngineXfader::kXfaderConfigKey, 
+               "xFaderMode"),this),
+
+m_CPXFaderReverse(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                  "xFaderReverse"),this)
+{
+    m_CPGuiTick.connectValueChanged(SLOT(slotGuiTick(double)));
+    startTimer(1000);
+}
+
+
+void ScrobblingManager::slotTrackPaused(TrackPointer pPausedTrack) {
+    Deck *sourceDeck = qobject_cast<Deck*>(sender());    
+    if (sourceDeck == 0) {
+        qDebug() << "Didn't load track in a deck yet scrobbling "
+                    "received paused signal.";
+        return;
+    }
+    QMutexLocker locker(&m_mutex);
+    QLinkedList<TrackInfo*>::Iterator it = m_trackList.begin();
+    while(it != m_trackList.end()) {
+        if ((*it)->m_pTrack == pPausedTrack and (*it)->m_pPlayer == sourceDeck) {
+            (*it)->m_trackInfo.pausePlayedTime();
+            break;
+        }
+        ++it;
+    }        
+}
+
+void ScrobblingManager::slotTrackResumed(TrackPointer pPausedTrack) {
+    Deck *sourceDeck = qobject_cast<Deck*>(sender());    
+    if (sourceDeck == 0) {
+        qDebug() << "Didn't load track in a deck yet scrobbling "
+                    "received resumed signal.";
+        return;
+    }    
+    if (isTrackAudible(pPausedTrack,sourceDeck)) {        
+        QMutexLocker locker(&m_mutex);
+        QLinkedList<TrackInfo*>::Iterator it = m_trackList.begin();
+        while(it != m_trackList.end()) {
+            if ((*it)->m_pTrack == pPausedTrack and (*it)->m_pPlayer == sourceDeck) {
```

Comment:
Why is the deck important here? 
Please add source code comments that explains this.

Reply:
So if a track plays in two decks, with significant volume, the timer counts twice as fast?